### PR TITLE
Remove ImageNet normalization for classify models

### LINKED ISSUE #1375: check_amp tries to download new weights
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

In trainer.py, l.624

`assert amp_allclose(YOLO('yolov8n.pt'), im)`

tries to download new weights, which is both inefficient and does not work in an offline setup (my case).

Solutions proposed:

1. Replace line 627 by `except (AssertionError, ConnectionError):`
2. Use input model weight path
3. Provide way to deactivate AMP check (right now check_amp is not even overload-able)

### Environment

- Ultralytics YOLOv8.0.52

### Minimal Reproducible Example

# Disable external web access
model = YOLO("insert_internal_path_for_weights_here")
results = model.train()  

### Additional

_No response_

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by glenn-jocher**
Experiment for removing input normalization to assist in better classification model exportability to iOS/Android apps.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Refactoring normalization parameters and improving AMP check handling.

### 📊 Key Changes
- Removed hard-coded ImageNet normalization in `autobackend.py` during image preprocessing.
- Made normalization parameters configurable by adding `mean` and `std` as arguments in `classify_transforms` function.
- Removed ImageNet normalization constants from `exporter.py` and modified the bias and scale calculation logic.
- Enhanced documentation and error handling for Automatic Mixed Precision (AMP) checks in `trainer.py`.

### 🎯 Purpose & Impact
- 🎨 The changes allow for more flexible image normalization, improving compatibility with different datasets beyond ImageNet.
- 🛠️ By parameterizing normalization, developers can fine-tune preprocessing based on specific use cases or domain needs.
- 🔍 The AMP check improvements ensure more reliable mixed precision training and helps avoid issues like NaN losses or zero-mAP results due to system anomalies.
- 🐛 Better error handling and documentation enhance the developer experience and model robustness for users.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@AyushExel @Laughing-q ok my results look good on this. I've retrained our cls models on Imagenet and gotten nearly identical results, so I'm going to approve this change and merge into updates for 8.0.53. 

The main difference here is that cls models will no longer include ImageNet normalization. It seems to bring no benefit (our det and seg models don't have this preprocessing step), and makes it harder for us to export to some formats like CoreML for our app.

This will be the first step in a longer campaign to attempt to align preprocessing between tasks, which some of our users have complained adds FLOPs to their workflows if they want to run classification and detection/segmentation on the same image.
